src/main/java/net/minecraftforge/common/extensions/IForgeLivingEntity.java

```diff
@@ -71,4 +71,15 @@ default boolean moveInFluid(FluidState state, Vec3 movementVector, double gravit
     {
         return state.move(self(), movementVector, gravity);
     }
+
+    /**
+     * Decides if this entity can disable shields, when it is attacking the {@code target}.
+     * @param target the entity which is attacked
+     * @return if this entity can disable shields
+     */
+    default boolean canDisableShield(LivingEntity target)
+    {
+        //noinspection deprecation
+        return self().canDisableShield() || self().getMainHandItem().canDisableShield(target.getUseItem(), target, self());
```

Comment:
This breaks the current contract of disabling shields. The method expands upon allowing a specific item to determine itself whether it can disable a shield. A check to the entity basically forces all axes to be able to disable shields, which may not be true.

In my opinion, the way to do this would be to have this redirect to the item and then have the warden override this method and either return true or the deprecated `#canDisableShield`.

Reply:
The fix you suggested would not account for future entities which customize the shield behaviour using `canDisableShield()Z`, needing each entity to have a patched-in override, which is not reliable, and does render the vanilla method useless, making it no longer have any effect.

Reply:
Sure, but there isn't really a good way to handle this since the original method does not contain the disabling target. Otherwise, I would suggest the method get redirected within the original method.